helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -433,6 +432,28 @@ public void removeDeploymentGroup(final String name) throws DeploymentGroupDoesN
     }
   }
 
+  private boolean updateOnHostChange(final DeploymentGroup group)
+      throws DeploymentGroupDoesNotExistException {
+    final DeploymentGroupStatus status = getDeploymentGroupStatus(group.getName());
+    if (status == null) {
+      // We're in an unknown state. Go hog wild.
+      return true;
+    }
+
+    if (group.getRollingUpdateReason() == null) {
+      // The last rolling update didn't fail, so there's no reason to expect this one will.
+      return (!status.getState().equals(FAILED));
+    }
+
+    if (group.getRollingUpdateReason().equals(HOSTS_CHANGED) && status.getState().equals(FAILED)) {
```

Comment:
Is that preferred? I knew you could, but thought for some reason I should default to `equals`.